GVFS/GVFS.Common/InstallerPreRunChecker.cs

```diff
@@ -0,0 +1,212 @@
+                consoleError = string.Join(
+                    Environment.NewLine, 
+                    "Blocking processes are running.",
+                    $"Run `{this.CommandToRerun}` again after quitting these processes - " + string.Join(", ", processList.ToArray()));
+                this.tracer.RelatedError($"{nameof(TryUnmountAllGVFSRepos)}: {consoleError}");
+                return false;
+            }
+
+            this.tracer.RelatedInfo("Successfully unmounted repositories.");
+
+            return true;
+        }
+
+        protected virtual bool IsElevated()
+        {
+            return GVFSPlatform.Instance.IsElevated();
+        }
+
+        protected virtual bool IsGVFSUpgradeSupported()
+        {
+            return GVFSPlatform.Instance.KernelDriver.IsGVFSUpgradeSupported();
+        }
+
+        protected virtual bool IsServiceInstalledAndNotRunning()
+        {
+            GVFSPlatform.Instance.IsServiceInstalledAndRunning(GVFSConstants.Service.ServiceName, out bool isInstalled, out bool isRunning);
+
+            return isInstalled && !isRunning;
+        }
+
+        protected virtual bool IsUnattended()
+        {
+            return GVFSEnlistment.IsUnattended(this.tracer);
+        }
+
+        protected virtual bool IsDevelopmentVersion()
+        {
+            return ProcessHelper.IsDevelopmentVersion();
+        }
+
+        protected virtual bool IsBlockingProcessRunning(out List<string> processes)
+        {
+            int currentProcessId = Process.GetCurrentProcess().Id;
+            Process[] allProcesses = Process.GetProcesses();
+            HashSet<string> matchingNames = new HashSet<string>();
+
+            foreach (Process process in allProcesses)
+            {
+                if (process.Id == currentProcessId || !BlockingProcessSet.Contains(process.ProcessName))
+                {
+                    continue;
+                }
+
+                matchingNames.Add(process.ProcessName);
+            }
+
+            processes = matchingNames.ToList();
+            return processes.Count > 0;
+        }
+
+        protected virtual bool TryRunGVFSWithArgs(string args, out string consoleError)
+        {
+            consoleError = null;
+
+            string gvfsPath = Path.Combine(
+                ProcessHelper.WhereDirectory(GVFSPlatform.Instance.Constants.GVFSExecutableName),
+                GVFSPlatform.Instance.Constants.GVFSExecutableName);
+
+            ProcessResult processResult = ProcessHelper.Run(gvfsPath, args);
+            if (processResult.ExitCode == 0)
+            {
+                return true;
+            }
+            else
+            {
+                string output = string.IsNullOrEmpty(processResult.Output) ? string.Empty : processResult.Output;
+                string errorString = string.IsNullOrEmpty(processResult.Errors) ? "GVFS error" : processResult.Errors;
+                consoleError = string.Format("{0}. {1}", errorString, output);

```

Comment:
This has been updated to show only stderr. Attaching latest screenshot.

<img width="518" alt="screen shot 2018-09-20 at 4 54 52 pm" src="https://user-images.githubusercontent.com/378580/45980280-8d2f3300-c01f-11e8-9986-c26c45b11635.png">

Reply:
The warning output looks good, but I have a couple of questions:

* What happened to the line about installing GVFS?
* Should we update the final line to something like "Even though some repos failed to mount, the upgrade completed successfully"? But more importantly, is there anything a user can do to recover here or is this just going to generate a support email?

Reply:
>> What happened to the line about installing GVFS?

It will be there in the actual release. This output is from my custom build. The latest release on GitHub/Release does not log mount failure list to stderr - you know. I commented out the code that installs GVFS upgrade, built and installed it with all the new mount-stderr related changes. Then ran gvfs upgrade. Now upgrade will download both Git and GVFS installers from Github/release, but will only install Git. Since the PC has the latest GVFS with mount-stderr changes, it shows the new stderr logging on the Console. 

>>Should we update the final line to something like "Even though some repos failed to mount, the upgrade completed successfully"?

I think that would break the existing pattern.
```
Progress spinners

ERROR/WARNING: Failure reason
Possible remedy in case of failure

[Upgrade completed successfully!] /*in case of success:*/
Press Enter to exit.
```

>>But more importantly, is there anything a user can do to recover here or is this just going to generate a support email?

It depends on the specific reason for the mount failure, right? We can't give a generic recovery step to cover all failure reasons. User can view the mount logs and take any recovery steps suggested there or email support. I'm not sure if there can be an alternate option.